tracing-opentelemetry/src/layer.rs

```diff
@@ -273,26 +218,17 @@ where
     /// // Get a tracer from the provider for a component
     /// let tracer = provider.get_tracer("component-name");
     ///
-    /// // The probability sampler can be used to export a percentage of spans
-    /// let sampler = sdk::Sampler::Probability(0.33);
-    ///
     /// // Create a layer with the configured tracer
-    /// let otel_layer = OpenTelemetryLayer::new(tracer, sampler);
+    /// let otel_layer = OpenTelemetryLayer::new(tracer);
     ///
     /// // Use the tracing subscriber `Registry`, or any other subscriber
     /// // that impls `LookupSpan`
-    /// let subscriber = Registry::default()
-    ///     .with(otel_layer);
+    /// let subscriber = Registry::default().with(otel_layer);
     /// # drop(subscriber);
     /// ```
-    pub fn new<Sampler>(tracer: T, sampler: Sampler) -> Self
-    where
-        Sampler: sdk::ShouldSample + 'static,
-    {
```

Comment:
This is a breaking API change --- does it make sense to leave the `new` constructor as-is, and add a new constructor that takes just a tracer, or does that not make sense?

Reply:
Could keep as is for a deprecation cycle, the sampler is no longer needed but the method could still accept one.

Reply:
Actually deprecation isn't really what we want here unless we want another name, but `with_tracer` already exists. Maybe best to just go with the breaking change as it is trivial to not include the parameter and it would only be called once during initialization.

Reply:
Okay, I'm fine with the breaking change if it's the best option.